packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -256,6 +254,10 @@ contract Bridge is EssentialContract, IBridge {
         whenNotPaused
         sameChain(message.destChainId)
     {
+        // TODO(Brecht): `message.owner`, but this is the `msg.sender` on the source chain.
+        // If the address is not owned by the same entity on the destination chain
+        // (e.g. can be the case for smart wallets/general contracts) this can give unexpected
+        // results (especially with refunding).
```

Comment:
Have to be careful with using address from the source chain on the destination chain! It really shouldn't be done.

Reply:
Seems it's indeed an issue.

Reply:
Lets fix it with new PRs.